src/server/svr_jobfunc.c

```diff
@@ -6805,19 +6820,17 @@ recreate_exec_vnode(job *pjob, char *vnodelist, char *err_msg,
 				}
 			}
 
-			if (((strcmp(pnode->nd_moms[0]->mi_host,
-						ms_fullhost) == 0) &&
-			     (pnode->nd_moms[0]->mi_port == ms_port)) ||
-			     ((vnodelist != NULL) &&	
+			if (is_parent_host_of_node(pnode, ms_fullhost, ms_port) ||
+			     ((vnodelist != NULL) &&
 			      !in_string_list(noden, '+', vnodelist))) {
```

Comment:
Can you combine the above two lines?  The line won't be much longer than the one above it.